pkg/deploy/reaper/reaper.go

```diff
@@ -65,6 +66,20 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, timeout time.
 			// Better not error out here...
 			glog.Infof("Cannot delete ReplicationController %s/%s: %v", rc.Namespace, rc.Name, err)
 		}
+
+		// Delete all deployer and hook pods
```

Comment:
Depending on the ration of complete+cancelled/failed deployments, you save a bunch of LIST calls for work that has already been done.